### PR TITLE
Remove sleep from p2p_network and conductor main.

### PR DESCRIPTION
**Pull request by struktured**
## PR summary

This PR addresses [waiting for p2p to be initalized](https://miro.com/app/board/o9J_kyiXmFs=/?moveToWidget=3074457346565169866) and [remove sleeps in the conductor](https://miro.com/app/board/o9J_kyiXmFs=/?moveToWidget=3074457346565169871.)

It takes advantage of PR https://github.com/holochain/holochain-rust/pull/1276 which will block the application until a `#P2P-Ready` signal is received (see https://github.com/holochain/holochain-rust/pull/1276#issuecomment-484273906). 

- removed sleep and loop statements from conductor/main.rs and p2p_network.rs.
- introduced shared channel between network handler and reducer init to publish and block on p2p ready event. panic if p2p is not ready within a timeout of 5 seconds. All of this logic is opaque to users of the `p2p_network`.
-  add `ThreadId(123)` to default logging format.
- introduced test to ensure reducer initialization completes with ok result.

## testing/benchmarking notes

- Added rust unit test to reducers/init.rs that verifies blocking behavior.

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

**TODO**
Please check one of the following, relating to the [CHANGELOG](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so there is an 'Unreleased' CHANGELOG item, with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: conductor/src/main.rs [hunk 1]
```diff
@@ -47,28 +47,23 @@ fn main() {
...
-            loop {
-                sleep(Duration::from_secs(1))
-            }
+            let mut conductor_guard = CONDUCTOR.lock().unwrap();
+            let mut conductor = conductor_guard.as_mut().expect("Conductor must be mounted");
+            println!(
+                "Successfully loaded {} instance configurations",
+                conductor.instances().len()
+            );
+            println!("Starting all of them...");
```
Comment (Connoropolous) [inline conductor/src/main.rs]:
```suggestion
            println!("Starting instances...");
```

**Comment by Connoropolous** [comment]
looking good @struktured

File: conductor/src/main.rs [hunk 2]
```diff
@@ -47,28 +47,23 @@ fn main() {
...
-                conductor.start_all_interfaces();
-                println!("Done.");
-                println!("Starting UI servers");
-                conductor
-                    .start_all_static_servers()
-                    .expect("Could not start UI servers!");
-            }
-            loop {
-                sleep(Duration::from_secs(1))
-            }
```
[the remark this answers is not in the document]
Comment (struktured) [reply conductor/src/main.rs]: @lucksus thanks for this explanation. I'll revert it back. Perhaps in a future PR we could wait for a sigint or sigkill instead which would eliminate the need for a sleep.

File: core/src/network/handler/mod.rs [hunk 3]
```diff
@@ -180,14 +182,17 @@ pub fn create_handler(c: &Arc<Context>, my_dna_address: String) -> NetHandler {
...
                 // Total hack in lieu of a world-model.
                 // Just republish everything when a new person comes on-line!!
                 republish_all_public_chain_entries(&context);
             }
             _ => {}
         }
         Ok(())
-    })
+    });
+    handler
```
[the remark this answers is not in the document]
Comment (struktured) [reply core/src/network/handler/mod.rs]: @lucksus no good reason- this code was modified originally to do the blocking but eventually taken out and placed in `p2p_network` instead. will undo this change.

File: conductor/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (struktured) [reply conductor/src/main.rs]: Done in b90fcdb8ba2c7225649a7be0c2556e860a5bb4fd

File: core/src/network/handler/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (struktured) [reply core/src/network/handler/mod.rs]: Done in ba0f618296f360e6c9a2f7ea7c185b89b9845e22.

**Comment by ddd-mtl** [comment]
@lucksus TweetLog is not active by default, so its normal not to see its output.

**Pull request merged**